TESTS/lorawan/application_api/main.cpp

```diff
@@ -0,0 +1,1389 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+
+#include "mbed.h"
+#if TARGET_MTS_MDOT_F411RE
+#include "SX1272_LoRaRadio.h"
+#endif
+
+#if defined(TARGET_K64F) || defined(TARGET_DISCO_L072CZ_LRWAN1)
+#include "SX1276_LoRaRadio.h"
+#endif
+#include "LoRaWANInterface.h"
+
+using namespace utest::v1;
+
+#define MBED_CONF_LORA_PHY 0
+
+#ifndef MBED_CONF_LORA_PHY
+#error "Must set LoRa PHY layer parameters."
+#else
+#if MBED_CONF_LORA_PHY      == 0
+ #include "LoRaPHYEU868.h"
+ static LoRaPHYEU868 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 1
+#include "LoRaPHYAS923.h"
+static LoRaPHYAS923 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 2
+#include "LoRaPHYAU915.h"
+static LoRaPHYAU915 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 3
+#include "LoRaPHYCN470.h"
+static LoRaPHYCN470 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 4
+#include "LoRaPHYCN779.h"
+static LoRaPHYCN779 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 5
+#include "LoRaPHYEU433.h"
+static LoRaPHYEU433 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 6
+#include "LoRaPHYIN865.h"
+static LoRaPHYIN865 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 7
+#include "LoRaPHYKR920.h"
+static LoRaPHYKR920 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 8
+#include "LoRaPHYUS915.h"
+static LoRaPHYUS915 lora_phy;
+#elif MBED_CONF_LORA_PHY    == 9
+#include "LoRaPHYUS915Hybrid.h"
+static LoRaPHYUS915Hybrid lora_phy;
+#else
+#error "Must set LoRa PHY layer parameters."
+#endif
+#endif
+
+#if defined(FEATURE_COMMON_PAL)
+#include "mbed_trace.h"
+#define TRACE_GROUP "LSTK"
+#else
+#define tr_debug(...) (void(0)) //dummies if feature common pal is not added
+#define tr_info(...)  (void(0)) //dummies if feature common pal is not added
+#define tr_error(...) (void(0)) //dummies if feature common pal is not added
+#define mbed_trace_mutex_wait_function_set(...) (void(0)) //dummies if feature common pal is not added
+#define mbed_trace_mutex_release_function_set(...) (void(0)) //dummies if feature common pal is not added
+#define mbed_trace_init(...) (void(0)) //dummies if feature common pal is not added
+#define mbed_trace_print_function_set(...) (void(0)) //dummies if feature common pal is not added
+#endif //defined(FEATURE_COMMON_PAL)
+
+Serial pc(USBTX, USBRX);
+static Mutex MyMutex;
+static void my_mutex_wait()
+{
+    MyMutex.lock();
+}
+static void my_mutex_release()
+{
+    MyMutex.unlock();
+}
+
+void trace_printer(const char* str)
+{
+    printf("%s\r\n", str);
+}
+
+void lora_event_handler(lora_events_t events);
+
+#if TARGET_MTS_MDOT_F411RE
+    SX1272_LoRaRadio Radio(LORA_MOSI, LORA_MISO, LORA_SCK, LORA_NSS, LORA_RESET,
+                           LORA_DIO0, LORA_DIO1, LORA_DIO2, LORA_DIO3, LORA_DIO4,
+                           LORA_DIO5, NC, NC, LORA_TXCTL, LORA_RXCTL, NC, NC);
+#endif
+#if TARGET_K64F
+    SX1276_LoRaRadio Radio(D11, D12, D13, D10, A0,
```

Comment:
K64F does not have defined pins LORA_MOSI, etc? Can be done via config? Same for disco below? to have one object in this file rather than ifdef else per target?

Same could be applied to specific header file per target ?